hack/verify-dist.sh

```diff
@@ -1,24 +1,87 @@
 #!/bin/bash
 
 set -e
+# show everything the shell is doing in case exit 1 isn't called on a fail
+# this is very noisy
+# set -o xtrace
 
-RED='\033[0;31m'
-NOCOLOR='\033[0m'
-GREEN='\033[0;32m'
+source "$(dirname "${BASH_SOURCE}")/lib/init.sh"
 
 # We don't need grunt to be installed globally for the system,  so
 # we can amend our path to look into the local node_modules for the
 # correct binaries.
 repo_root="$( dirname "${BASH_SOURCE}" )/.."
 export PATH="${PATH}:${repo_root}/node_modules/grunt-cli/bin"
 
+# vars to generate a fake junit test output file
+TIMESTAMP=$(date +"%x %r %Z")
+SUITE_NAME="$BASH_SOURCE/$FILE"
+TEST_NAME="User should generate /dist by running grunt build"
+TESTS_COUNT=1
+SKIPPED_COUNT=0
+DISABLED=0
+FAILURE_COUNT=0
+ERROR_COUNT=0
+PASSED=false
+POTENTIAL_FAILED_TEST=""
+
+# run the build to generate dist
+os::log::info "Rebuilding dist..."
 grunt build
 
-echo "Verifying that checked in built files under dist match the source..."
-if [[ $(git status -s -u dist*) ]]; then
-    echo -e "${RED}Built /dist does not match what is committed, run 'grunt build' and include the results in your commit.${NOCOLOR}"
-    git diff --exit-code dist*
-    exit 1
+# validate generated against what is committed
+os::log::info "Verifying that checked in built files under dist match the source..."
+GIT_STATUS="$( git status --short --untracked-files -- dist/ )"
+GIT_DIFF="$( git diff )"
+
+# -n = not null
+if [[ -n "$GIT_STATUS" ]]; then
+  FAILURE_COUNT=1
+  FAILURE_MESSAGE="Built /dist does not match what is committed, run grunt build and include the results in your commit."
+  ERROR_COUNT=1
+  POTENTIAL_FAILED_TEST="<failure type=\"DIST_FAILURE\" message=\"$FAILURE_MESSAGE\"><![CDATA[ $GIT_DIFF ]]></failure>"
```

Comment:
@stevekuznetsov so long as this `CDATA` block isn't objectionable, I think I'm set on this PR & can stop bothering you about bash for a bit.  :stuck_out_tongue: